pkg/agent/nodeportlocal/npl_agent_init.go

```diff
@@ -17,44 +17,157 @@
 package nodeportlocal
 
 import (
+	"context"
+	"encoding/json"
 	"errors"
 	"fmt"
 	"time"
 
 	nplk8s "github.com/vmware-tanzu/antrea/pkg/agent/nodeportlocal/k8s"
 	"github.com/vmware-tanzu/antrea/pkg/agent/nodeportlocal/portcache"
+	"github.com/vmware-tanzu/antrea/pkg/agent/nodeportlocal/rules"
 	"github.com/vmware-tanzu/antrea/pkg/agent/nodeportlocal/util"
 
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/informers"
 	coreinformers "k8s.io/client-go/informers/core/v1"
 	clientset "k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
 )
 
 // Set resyncPeriod to 0 to disable resyncing.
 // UpdateFunc event handler will be called only when the object is actually updated.
 const resyncPeriod = 0 * time.Minute
 
+func cleanupNPLAnnotationForPod(kubeClient clientset.Interface, pod corev1.Pod) error {
+	type patchStringValue struct {
+		Op    string            `json:"op"`
+		Path  string            `json:"path"`
+		Value map[string]string `json:"value"`
+	}
+
+	_, ok := pod.Annotations[nplk8s.NPLAnnotationKey]
+	if !ok {
+		return nil
+	}
+	annotations := pod.Annotations
+	delete(annotations, nplk8s.NPLAnnotationKey)
+	payload := []patchStringValue{
```

Comment:
I think we can delete a single annotation by setting the annotation to nil.

Could you change the annotation update/setting code with patching too?

Reply:
From: https://tools.ietf.org/html/rfc7386
```
Changing the value of "a" and removing "f" can be achieved by
   sending:
       PATCH /target HTTP/1.1
       Host: example.org
       Content-Type: application/merge-patch+json
       {
         "a":"z",
         "c": {
       "f": null
         }
       }
   When applied to the target resource, the value of the "a" member is
   replaced with "z" and "f" is removed, leaving the remaining content
   untouched.
```

Reply:
```
k8sClient.CoreV1().Pods("default").Patch(context.TODO(), "pod1", types.MergePatchType, []byte(`{"metadata":{"annotations":{"abc":null}}}`), metav1.PatchOptions{})
```
This can remove annotation "abc".

Reply:
> This can remove annotation "abc".

Yes, this works. Thanks! Have updated the PR.

> Could you change the annotation update/setting code with patching too?

Done.